test/e2e/kubectl.go

```diff
@@ -0,0 +1,201 @@
+*/
+
+package e2e
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+)
+
+const (
+	nautilusImage       = "kubernetes/update-demo:nautilus"
+	kittenImage         = "kubernetes/update-demo:kitten"
+	updateDemoSelector  = "name=update-demo"
+	updateDemoContainer = "update-demo"
+)
+
+var _ = Describe("kubectl", func() {
+
+	updateDemoRoot := filepath.Join(root, "examples/update-demo")
+	nautilusPath := filepath.Join(updateDemoRoot, "nautilus-rc.yaml")
+	kittenPath := filepath.Join(updateDemoRoot, "kitten-rc.yaml")
+
+	It("should create and stop a replication controller", func() {
+		defer cleanup(nautilusPath)
+
+		By("creating a replication controller")
+		runKubectl("create", "-f", nautilusPath)
+		validateController(nautilusImage, 2, 30*time.Second)
+	})
+
+	It("should scale a replication controller", func() {
+		defer cleanup(nautilusPath)
+
+		By("creating a replication controller")
+		runKubectl("create", "-f", nautilusPath)
+		validateController(nautilusImage, 2, 30*time.Second)
+		By("scaling down the replication controller")
+		runKubectl("resize", "rc", "update-demo-nautilus", "--replicas=1")
+		validateController(nautilusImage, 1, 30*time.Second)
+		By("scaling up the replication controller")
+		runKubectl("resize", "rc", "update-demo-nautilus", "--replicas=2")
+		validateController(nautilusImage, 2, 30*time.Second)
+	})
+
+	It("should do a rolling update of a replication controller", func() {
+		// Cleanup all resources incase we fail somewhere in the middle
+		defer cleanup(updateDemoRoot)
+
+		By("creating the initial replication controller")
+		runKubectl("create", "-f", nautilusPath)
+		validateController(nautilusImage, 2, 30*time.Second)
+		By("rollingupdate to new replication controller")
+		runKubectl("rollingupdate", "update-demo-nautilus", "--update-period=1s", "-f", kittenPath)
+		validateController(kittenImage, 2, 30*time.Second)
+	})
+
+})
+
+func cleanup(filePath string) {
+	By("using stop to clean up resources")
+	runKubectl("stop", "-f", filePath)
+
+	resources := runKubectl("get", "pods,rc", "-l", updateDemoSelector, "--no-headers")
+	if resources != "" {
+		Failf("Resources left running after stop:\n%s", resources)
+	}
+}
+
+func validateController(image string, replicas int, timeout time.Duration) {
+
+	getPodsTemplate := "--template={{range.items}}{{.id}} {{end}}"
+
+	// NB: kubectl adds the "exists" function to the standard template functions.
+	// This lets us check to see if the "running" entry exists for each of the containers
+	// we care about. Exists will never return an error and it's safe to check a chain of
+	// things, any one of which may not exist. In the below template, all of info,
+	// containername, and running might be nil, so the normal index function isn't very
+	// helpful.
+	// This template is unit-tested in kubec{tl|fg}, so if you change it, update the unit test.
```

Comment:
nit: Since there's no longer a kubecfg, you can replace kubec{tl|fg} with the much easier to read kubectl